nomenclature/region_mapping_models.py

```diff
@@ -241,29 +248,38 @@ def apply(self, df: IamDataFrame) -> IamDataFrame:
 
                 # Aggregate
                 if self.mappings[model].common_regions is not None:
-                    for cr in self.mappings[model].common_regions:
-                        PYAM_AGG_KWARGS = {
-                            "components",
-                            "method",
-                            "weight",
-                            "drop_negative_weights",
+                    vars = {
+                        var: {
```

Comment:
for readability, I'd pull that out into an own utility function, e.g., `_filter_dict_args(kwargs, keys)` with `keys` defaulting to  `PYAM_AGG_KWARGS`

Reply:
Done